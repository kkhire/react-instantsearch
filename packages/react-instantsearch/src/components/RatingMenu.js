import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { isEmpty } from 'lodash';

import translatable from '../core/translatable';

class RatingMenu extends Component {
  static propTypes = {
    cx: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    refine: PropTypes.func.isRequired,
    createURL: PropTypes.func.isRequired,
    min: PropTypes.number,
    max: PropTypes.number,
    currentRefinement: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    count: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.string,
        count: PropTypes.number,
      })
    ),
    canRefine: PropTypes.bool.isRequired,
  };

  onClick(min, max, e) {
    e.preventDefault();
    e.stopPropagation();
    if (
      min === this.props.currentRefinement.min &&
      max === this.props.currentRefinement.max
    ) {
      this.props.refine({ min: this.props.min, max: this.props.max });
    } else {
      this.props.refine({ min, max });
    }
  }

  buildItem({
    cx,
    max,
    lowerBound,
    count,
    translate,
    createURL,
    isLastSelectableItem,
  }) {
    const disabled = !count;
    const isCurrentMinLower = this.props.currentRefinement.min < lowerBound;
    const selected =
      (isLastSelectableItem && isCurrentMinLower) ||
      (!disabled &&
        lowerBound === this.props.currentRefinement.min &&
        max === this.props.currentRefinement.max);

    const icons = [];
    let rating = 0;
    for (let icon = 0; icon < max; icon++) {
      if (icon < lowerBound) {
        rating++;
      }
      icons.push([
        <svg
          key={icon}
          className={cx(
            'starIcon',
            icon >= lowerBound ? 'starIcon--empty' : 'starIcon--full'
          )}
          aria-hidden="true"
          width="24"
          height="24"
        >
          <use
            xlinkHref={`#${cx(
              icon >= lowerBound ? 'starEmptySymbol' : 'starSymbol'
            )}`}
          />
        </svg>,
        ' ',
      ]);
    }

    // The last item of the list (the default item), should not
    // be clickable if it is selected.
    const onClickHandler =
      !disabled && (!isLastSelectableItem || !selected)
        ? {
            href: createURL({ min: lowerBound, max }),
            onClick: this.onClick.bind(this, lowerBound, max),
          }
        : {};

    return (
      <li
        key={lowerBound}
        className={cx(
          'item',
          selected && 'item--selected',
          disabled && 'item--disabled'
        )}
      >
        <a
          aria-label={`${rating}${translate('ratingLabel')}`}
          className={cx('link')}
          {...onClickHandler}
        >
          {icons}
          <span className={cx('label')} aria-hidden="true">
            {translate('ratingLabel')}
          </span>{' '}
          <span className={cx('count')}>{count}</span>
        </a>
      </li>
    );
  }

  render() {
    const {
      cx,
      translate,
      refine,
      min,
      max,
      count,
      createURL,
      canRefine,
    } = this.props;
    const items = [];
    for (let i = max; i >= min; i--) {
      const hasCount = !isEmpty(count.filter(item => Number(item.value) === i));
      const lastSelectableItem = count.reduce(
        (acc, item) =>
          item.value < acc.value || (!acc.value && hasCount) ? item : acc,
        {}
      );
      const itemCount = count.reduce(
        (acc, item) => (item.value >= i && hasCount ? acc + item.count : acc),
        0
      );
      items.push(
        this.buildItem({
          cx,
          lowerBound: i,
          max,
          refine,
          count: itemCount,
          translate,
          createURL,
          isLastSelectableItem: i === Number(lastSelectableItem.value),
        })
      );
    }
    return (
      <div>
        <svg xmlns="http://www.w3.org/2000/svg" style={{ display: 'none' }}>
          <symbol id={cx('starSymbol')} viewBox="0 0 24 24">
            <path d="M12 .288l2.833 8.718h9.167l-7.417 5.389 2.833 8.718-7.416-5.388-7.417 5.388 2.833-8.718-7.416-5.389h9.167z" />
          </symbol>
          <symbol id={cx('starEmptySymbol')} viewBox="0 0 24 24">
            <path d="M12 6.76l1.379 4.246h4.465l-3.612 2.625 1.379 4.246-3.611-2.625-3.612 2.625 1.379-4.246-3.612-2.625h4.465l1.38-4.246zm0-6.472l-2.833 8.718h-9.167l7.416 5.389-2.833 8.718 7.417-5.388 7.416 5.388-2.833-8.718 7.417-5.389h-9.167l-2.833-8.718z" />
          </symbol>
        </svg>
        <ul className={cx('list', !canRefine && 'list--noRefinement')}>
          {items}
        </ul>
      </div>
    );
  }
}

export default translatable({
  ratingLabel: ' & Up',
})(RatingMenu);
