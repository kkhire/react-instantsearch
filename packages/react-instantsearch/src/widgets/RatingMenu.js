import React from 'react';
import PropTypes from 'prop-types';
import connectRange from '../connectors/connectRange';
import classNames from '../components/classNames';
import BaseWidget from '../components/BaseWidget';
import RatingMenuComponent from '../components/RatingMenu';

const cx = classNames('RatingMenu');

/**
 * RatingMenu lets the user refine search results by clicking on stars.
 *
 * The stars are based on the selected `attributeName`.
 * @requirements The attribute passed to the `attributeName` prop must be holding numerical values.
 * @name RatingMenu
 * @kind widget
 * @propType {string} attributeName - the name of the attribute in the record
 * @propType {number} [min] - Minimum value for the rating. When this isn't set, the minimum value will be automatically computed by Algolia using the data in the index.
 * @propType {number} [max] - Maximum value for the rating. When this isn't set, the maximum value will be automatically computed by Algolia using the data in the index.
 * @propType {{min: number, max: number}} [defaultRefinement] - Default state of the widget containing the lower bound (end) and the max for the rating.
 * @propType {function} [renderHeader] - Adds a header to the widget.
 * @propType {function} [renderFooter] - Adds a footer to the widget.
 * @themeKey ais-RatingMenu - the root div of the widget
 * @themeKey ais-RatingMenu-header - the header of the widget (optional)
 * @themeKey ais-RatingMenu-body - the body of the widget
 * @themeKey ais-RatingMenu-list - the list of ratings
 * @themeKey ais-RatingMenu-item - the rating list item
 * @themeKey ais-RatingMenu-item--selected - the selected rating list item
 * @themeKey ais-RatingMenu-item--disabled - the disabled rating list item
 * @themeKey ais-RatingMenu-link - the rating clickable item
 * @themeKey ais-RatingMenu-starIcon--full - the filled star icon
 * @themeKey ais-RatingMenu-starIcon--empty - the empty star icon
 * @themeKey ais-RatingMenu-starIcon - the star icon
 * @themeKey ais-RatingMenu-label - the label used after the stars
 * @themeKey ais-RatingMenu-count - the count of ratings for a specific item
 * @themeKey ais-RatingMenu-footer - the footer of the widget (optional)
 * @translationKey ratingLabel - Label value for the rating link
 * @example
 * import React from 'react';
 *
 * import { RatingMenu, InstantSearch } from 'react-instantsearch/dom';
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <RatingMenu attributeName="rating" />
 *     </InstantSearch>
 *   );
 * }
 */

const Widget = ({ canRefine, renderHeader, renderFooter, ...props }) => (
  <BaseWidget
    cx={cx}
    canRefine={canRefine}
    renderHeader={renderHeader}
    renderFooter={renderFooter}
  >
    <RatingMenuComponent {...props} cx={cx} canRefine={canRefine} />
  </BaseWidget>
);

Widget.propTypes = {
  canRefine: PropTypes.bool.isRequired,
  renderHeader: PropTypes.func,
  renderFooter: PropTypes.func,
};

export default connectRange(Widget);
