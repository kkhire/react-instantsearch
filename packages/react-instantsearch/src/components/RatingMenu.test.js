/* eslint-env jest, jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import RatingMenu from './RatingMenu';

describe('RatingMenu', () => {
  it('supports passing max/min values', () => {
    const tree = renderer
      .create(
        <RatingMenu
          cx={(...x) => x.join(' ')}
          createURL={() => '#'}
          refine={() => null}
          min={1}
          max={5}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies translations', () => {
    const tree = renderer
      .create(
        <RatingMenu
          cx={(...x) => x.join(' ')}
          createURL={() => '#'}
          refine={() => null}
          translations={{
            ratingLabel: ' & Up',
          }}
          min={1}
          max={5}
          currentRefinement={{ min: 1, max: 5 }}
          count={[
            { value: '1', count: 1 },
            { value: '2', count: 2 },
            { value: '3', count: 3 },
            { value: '4', count: 4 },
            { value: '5', count: 5 },
          ]}
          canRefine={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  const refine = jest.fn();
  const createURL = jest.fn();
  const starRating = (
    <RatingMenu
      cx={(...x) => x.join(' ')}
      createURL={createURL}
      refine={refine}
      min={1}
      max={5}
      currentRefinement={{ min: 1, max: 5 }}
      count={[
        { value: '1', count: 4 },
        { value: '2', count: 2 },
        { value: '3', count: 3 },
        { value: '4', count: 3 },
        { value: '5', count: 0 },
      ]}
      canRefine={true}
    />
  );

  beforeEach(() => {
    refine.mockClear();
    createURL.mockClear();
  });

  it('should create an URL for each row except for the largest: the default selected one', () => {
    const wrapper = mount(starRating);

    expect(createURL.mock.calls).toHaveLength(3);
    expect(createURL.mock.calls[0][0]).toEqual({ min: 4, max: 5 });
    expect(createURL.mock.calls[1][0]).toEqual({ min: 3, max: 5 });
    expect(createURL.mock.calls[2][0]).toEqual({ min: 2, max: 5 });

    wrapper.unmount();
  });

  it('refines its value on change', () => {
    const wrapper = mount(starRating);
    const links = wrapper.find('.link');
    expect(links).toHaveLength(5);

    let selectedItem = wrapper.find('.item--selected');
    expect(selectedItem).toHaveLength(1);

    links.first().simulate('click');

    expect(refine.mock.calls).toHaveLength(0);

    selectedItem = wrapper.find('.item--selected');
    expect(selectedItem).toBeDefined();

    const disabledIcon = wrapper.find('.item--disabled').find('.starIcon');

    expect(disabledIcon).toHaveLength(5);
    wrapper.unmount();
  });

  it('should display the right number of stars', () => {
    const wrapper = mount(starRating);
    wrapper
      .find('.link')
      .last()
      .simulate('click');

    const selectedItem = wrapper.find('.item--selected');

    const fullIcon = selectedItem.find('.starIcon--full');
    const emptyIcon = selectedItem.find('.starIcon--empty');

    expect(fullIcon).toHaveLength(1);
    expect(emptyIcon).toHaveLength(4);
    wrapper.unmount();
  });

  it('clicking on the selected refinement should select the largest range', () => {
    const wrapper = mount(starRating);
    wrapper.setProps({ currentRefinement: { min: 4, max: 5 } });

    const links = wrapper.find('.link');
    links.at(1).simulate('click');

    expect(refine.mock.calls).toHaveLength(1);
    expect(refine.mock.calls[0][0]).toEqual({ min: 1, max: 5 });
    wrapper.unmount();
  });

  it('the default selected range should be the lowest one with count', () => {
    const wrapper = mount(starRating);
    wrapper.setProps({
      count: [
        { value: '2', count: 2 },
        { value: '3', count: 3 },
        { value: '4', count: 3 },
      ],
    });

    const items = wrapper.find('.item');
    expect(items.first().hasClass('item--selected')).toBe(false);

    const selected = wrapper.find('.item--selected');
    expect(selected.find('.starIcon--empty')).toHaveLength(3);
    expect(selected.find('.starIcon--full')).toHaveLength(2);
    expect(selected.text()).toContain('8');

    wrapper.unmount();
  });
});
