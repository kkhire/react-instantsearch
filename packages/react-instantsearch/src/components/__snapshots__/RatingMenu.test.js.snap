// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingMenu applies translations 1`] = `
<div>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="starSymbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 .288l2.833 8.718h9.167l-7.417 5.389 2.833 8.718-7.416-5.388-7.417 5.388 2.833-8.718-7.416-5.389h9.167z"
      />
    </symbol>
    <symbol
      id="starEmptySymbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 6.76l1.379 4.246h4.465l-3.612 2.625 1.379 4.246-3.611-2.625-3.612 2.625 1.379-4.246-3.612-2.625h4.465l1.38-4.246zm0-6.472l-2.833 8.718h-9.167l7.416 5.389-2.833 8.718 7.417-5.388 7.416 5.388-2.833-8.718 7.417-5.389h-9.167l-2.833-8.718z"
      />
    </symbol>
  </svg>
  <ul
    className="list false"
  >
    <li
      className="item false false"
    >
      <a
        aria-label="5 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          5
        </span>
      </a>
    </li>
    <li
      className="item false false"
    >
      <a
        aria-label="4 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          9
        </span>
      </a>
    </li>
    <li
      className="item false false"
    >
      <a
        aria-label="3 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          12
        </span>
      </a>
    </li>
    <li
      className="item false false"
    >
      <a
        aria-label="2 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          14
        </span>
      </a>
    </li>
    <li
      className="item item--selected false"
    >
      <a
        aria-label="1 & Up"
        className="link"
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          15
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`RatingMenu supports passing max/min values 1`] = `
<div>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="starSymbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 .288l2.833 8.718h9.167l-7.417 5.389 2.833 8.718-7.416-5.388-7.417 5.388 2.833-8.718-7.416-5.389h9.167z"
      />
    </symbol>
    <symbol
      id="starEmptySymbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 6.76l1.379 4.246h4.465l-3.612 2.625 1.379 4.246-3.611-2.625-3.612 2.625 1.379-4.246-3.612-2.625h4.465l1.38-4.246zm0-6.472l-2.833 8.718h-9.167l7.416 5.389-2.833 8.718 7.417-5.388 7.416 5.388-2.833-8.718 7.417-5.389h-9.167l-2.833-8.718z"
      />
    </symbol>
  </svg>
  <ul
    className="list false"
  >
    <li
      className="item false false"
    >
      <a
        aria-label="5 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          5
        </span>
      </a>
    </li>
    <li
      className="item false false"
    >
      <a
        aria-label="4 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          9
        </span>
      </a>
    </li>
    <li
      className="item false false"
    >
      <a
        aria-label="3 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          12
        </span>
      </a>
    </li>
    <li
      className="item false false"
    >
      <a
        aria-label="2 & Up"
        className="link"
        href="#"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          14
        </span>
      </a>
    </li>
    <li
      className="item item--selected false"
    >
      <a
        aria-label="1 & Up"
        className="link"
      >
        <svg
          aria-hidden="true"
          className="starIcon starIcon--full"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starSymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <svg
          aria-hidden="true"
          className="starIcon starIcon--empty"
          height="24"
          width="24"
        >
          <use
            xlinkHref="#starEmptySymbol"
          />
        </svg>
         
        <span
          aria-hidden="true"
          className="label"
        >
           & Up
        </span>
         
        <span
          className="count"
        >
          15
        </span>
      </a>
    </li>
  </ul>
</div>
`;
